### PR TITLE
[PF-1554] terra-cli: gradle and java version upgrades

### PR DESCRIPTION
**Pull request by dexamundsen**
Below changes are made to keep this consistent with other services

- Gradle upgrade from 6.x to 7.4.1
- JDK upgrade from 11 to 17
- Gradle build cleanup (based on https://github.com/DataBiosphere/terra-common-lib/pull/26 and other Terra repos)

Subsequent changes that were required

- Update Docker image used to JDK17
- Update documentation
- Update library (dependency) versions and code fixes to align with these changes
- Minor change to Start up script to fix earlier error with gradle upgrade

notebook JDK update - https://github.com/DataBiosphere/terra-workspace-manager/pull/776

Pre-release used to verily nightly test - https://github.com/DataBiosphere/terra-cli/releases/tag/dexamundsen-1932

Nightly test - (run on modified changeset with updated version number) 
https://github.com/DataBiosphere/terra-cli/actions/runs/3048022035

---
Root cause of nighty test failures during earlier attempts of gradle upgrade - 

Error msg - Could not find / load main class x.x.x.... caused due to noClassDef ...
Error root cause - the new version of gradle application plugin changed the start scripts that were auto-generated. Hence our custom code to update the APP_HOME in build.gradle and subsequently the CLASSPATH were incorrectly set.
Error fix - Update our custom code that sets APP_HOME in the build file

### CODE REVIEW

File: tools/build-docker.sh [hunk 1]
```diff
@@ -30,7 +30,8 @@ fi
 
 echo "Building the image"
 localImageNameAndTag="$localImageName:$localImageTag"
-docker build -t $localImageNameAndTag ./docker
+targetPlatform="--platform=linux/amd64"
```
Comment (melissachang) [inline tools/build-docker.sh:33]: Add comment on why you're setting this

**Review by melissachang — APPROVED** [review]
Nightly tests passed: https://github.com/DataBiosphere/terra-cli/actions/runs/3048022035

File: build.gradle [hunk 2]
```diff
@@ -1,39 +1,33 @@
...
+// If true, search local repository (~/.m2/repository/) first for dependencies.
+def useMavenLocal = false
 repositories {
+    if (useMavenLocal) {
+        mavenLocal() // must be listed first to take effect
+    }
     mavenCentral()
     maven {
-        url 'https://broadinstitute.jfrog.io/broadinstitute/libs-release-local/'
+        url 'https://broadinstitute.jfrog.io/broadinstitute/libs-release/'
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply build.gradle]: Both end up referring to the same, and this change would have no effect. However, the artifacts repo explicitly calls them *-local (https://broadinstitute.jfrog.io/ui/repos/tree/General/) I will retain the -local suffix.

**Review by zloery — APPROVED** [review]

File: tools/build-docker.sh (hunk 1 above)
Comment (dexamundsen) [reply tools/build-docker.sh:34]: done

**Review by dexamundsen — COMMENTED** [review]
Addressed comments, done

File: .gitignore [hunk 3]
```diff
@@ -42,6 +42,3 @@ out/
 ### Nextflow
 .nextflow
 .nextflow.log
-
-### Jenv directory-local version setting
-.java-version
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply .gitignore:47]: sure

**Pull request merged**